src/input/agent.rs

```diff
@@ -124,13 +126,31 @@ where
             }
         };
 
+        // Parse decision rule
+        let decision_rule = match agent_raw.decision_rule.as_str() {
+            "single" => DecisionRule::Single,
+            "weighted" => DecisionRule::Weighted,
+            "lexico" => {
+                let tolerance = agent_raw
+                    .decision_lexico_tolerance
+                    .ok_or_else(|| anyhow::anyhow!("Missing tolerance for lexico decision rule"))?;
+                ensure!(
+                    tolerance >= 0.0,
+                    "Lexico tolerance must be non-negative, got {}",
+                    tolerance
+                );
+                DecisionRule::Lexicographical { tolerance }
+            }
+            invalid_rule => return Err(anyhow::anyhow!("Invalid decision rule: {}", invalid_rule)),
```

Comment:
You can use the `bail!` macro here:

```suggestion
            invalid_rule => bail!("Invalid decision rule: {invalid_rule}"))
```